### ISSUE TITLE
Nekonzistence dat RÚIAN???

### ISSUE THREAD

**Issue by raugustyn** [vugtk21/RUIANToolbox#179]
File "C:\Users\raugustyn\Desktop\RUIAN\RUIANToolbox\RUIANDownloader\RUIANDownload.py", line 519, in <module>
    main()
  File "C:\Users\raugustyn\Desktop\RUIAN\RUIANToolbox\RUIANDownloader\RUIANDownload.py", line 515, in main
    doImport(argv)
  File "C:\Users\raugustyn\Desktop\RUIAN\RUIANToolbox\RUIANImporter\importRUIAN.py", line 290, in doImport
    buildAll()
  File "C:\Users\raugustyn\Desktop\RUIAN\RUIANToolbox\RUIANServices\services\auxiliarytables.py", line 262, in buildAll
    buildAutocompleteTables()
  File "C:\Users\raugustyn\Desktop\RUIAN\RUIANToolbox\RUIANServices\services\auxiliarytables.py", line 258, in buildAutocompleteTables
    buildGIDsTable()
  File "C:\Users\raugustyn\Desktop\RUIAN\RUIANToolbox\RUIANServices\services\auxiliarytables.py", line 210, in buildGIDsTable
    houseNumber, recordNumber = HTTPShared.analyseRow(typ_so, houseNumber)
  File "C:\Users\raugustyn\Desktop\RUIAN\RUIANToolbox\RUIANServices\services\HTTPShared.py", line 733, in analyseRow
    if typ_so[-3:] == ".p.":

**Comment by raugustyn** [comment]
Je tam podivný záznam 73315800;;"";"";"";;"";"Větrná";"";159;;"";26701;774951.93;1054977.02;"2015-02-05T00:00:00";"01010000208A150000C3F528DC4FA627C152B81E05011930C1"

**Comment by raugustyn** [comment]
Podle seznam.cz je to asi:
Větrná 159
267 01  Trubín
okres Beroun, kraj Středočeský

**Comment by vacektom** [comment]
Tak podle daného ID RÚIAN 73315800 je adresa opravdu jiná, konkrétně:
č.p. 159
26701 Trubín
Na tento nesoulad jste přišel jakým způsobem, že tam v datech je tento záznam? Při importu? Podle mě v datech RÚIAN mohou být tyto nesoulady ve větší míře, 100% čistá, tato data určitě nebudou...

**Comment by vacektom** [comment]
Toto ověření platí i o našich službách, kde to výše uvedenou adresu v databázi také nalezne.

**Comment by raugustyn** [comment]
Jako že jsem vče updatoval databázi na svém notebooku abych ověřil vytvoření tabulky pro #174 a vyhodilo mi to tuhle chybovou hlášku.

Dal jsem si jenom select * from address_points where typ_so is null a vyjel mi tenhle záznam.

**Comment by vacektom** [comment]
Podle VDP je adresa s ID RÚIAN 73315800 následující:
Větrná 159
 267 01 Trubín
Což znamená, že se v našich službách vyskytuje jiná adresa, než skutečně existuje. Jde o to, zda je chyba ve stahovaných souborech VDP, nebo se data nestáhnou pomocí Downloaderu a Importeru korektně. Zkrátka tam chybí název té ulice, což je špatně.

**Comment by raugustyn** [comment]
Ono to generování adres posle vyhlášky pro našeptávač Id je asi zbytečné, v tom má souček pravdu. Ale na druhou stranu to fakt projde záznam po záznamu, sestaví adresu a uloží si ji, tzn. když tam je nějaká faktická nesrovnalost, tak jí najde.

**Comment by vacektom** [comment]
Napadá mě jen následující věc, jak to ověřit. Stáhnout si z VDP soubor dané obce a vyhledat podle ID danou adresu, zda v daném souboru není chyba. Pokud ano, je to chyba RÚIAN, pokud ne, nějakým způsobem se ztratí název té adresy, což mi připadá divné.

**Comment by raugustyn** [comment]
Stahovaly se tyhle patche, v nich by to mělo někde bejt:
http://vdp.cuzk.cz/vymenny_format/soucasna/20150217_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150216_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150215_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150213_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150212_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150211_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150210_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150209_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150208_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150207_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150206_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150205_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150204_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150203_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150202_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150201_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150131_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150130_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150129_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150128_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150127_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150126_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150125_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150124_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150123_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150122_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150121_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150120_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150119_ST_ZKSH.xml.gz
http://vdp.cuzk.cz/vymenny_format/soucasna/20150118_ST_ZKSH.xml.gz

**Comment by vacektom** [comment]
Takže chyba bude patrně někde ve stahovaných aktualizacích, jestli to dobře chápu...

**Comment by raugustyn** [comment]
Jo, v tom záznamu je sice RUIAN id, číslo popisné, psč, ulice a souřadnice, ale není tam název obce, typ_so apod.

**Comment by raugustyn** [comment]
V datech aktualizovaných 15.2. tam je to ještě oky:
http://www.vugtk.cz/euradin/AddressDB/rest.py/CompileAddress/text?AddressPlaceId=73315800&ExtraInformation=standard
